src/log/io_backend.cr

```diff
@@ -3,16 +3,20 @@ class Log::IOBackend < Log::Backend
   property io : IO
   property formatter : Formatter
 
-  def initialize(@io = STDOUT, *, @formatter : Formatter = ShortFormat)
-    @mutex = Mutex.new(:unchecked)
+  {% if flag?(:win32) %}
+    private DEFAULT_DISPATCHER = DispatchMode::Sync
+  {% else %}
+    private DEFAULT_DISPATCHER = DispatchMode::Async
+  {% end %}
+
+  def initialize(@io = STDOUT, *, @formatter : Formatter = ShortFormat, dispatcher : Dispatcher::Spec? = nil)
```

Comment:
It bothers me a bit that we need to accept `nil` as an argument only to allow a different default in Windows.

How about defining an initialize without dispatcher argument that will compute the default one and leave the dispatcher argument without `nil` in this one?

Eventually if the default is the same in all platforms, we will be able to simplify the code as we would've wanted in the first place without changing the actual interface.

Reply:
TBH, it was nilable before I did the change to fix the issue in Windows. I'll update with the change suggested by @asterite. The problem is that `DEFAULT_DISPATCHER` is private but it will show up in docs. I don't want to make it public either because that should be deleted once async works in Windows.

Reply:
To hide the default, you can just add a `self.new` overload without `dispatcher` argument.

Reply:
I guess you can add one initialize or the other depending on the flag.

Reply:
Honestly I'm not happy having to add all this stuff to workaround issues in Windows specs. The Windows version even though it has received significant and amazing progress is still not usable if things like channels or sockets doesn't work. It should not make us shape the design of everything else. To me the best option is disable the specs that doesn't work until the basic functionality is ready.

Reply:
> having to add all this stuff

Do you mean just one `{% if flag?(:win32) %}`?

Reply:
No, I mean making the logger to use the sync dispatcher in this case, because the async doesn't work. Is not that the async couldn't possible work in Windows and thus we need to add a different default. It's a temporary issue until the underlying problem with channels is fixed. What's the point of making this initialization different just because the Windows implementation is incomplete. Who's going to use this in real life anyway?

Reply:
I've seen @oprypin use Crystal to create games on Windows. If the sync logger works by default in Windows then I think we should leave it like that.

Or put another way, what's the problem with generating two `initialize`, one for non-Windows and one for Windows?

Reply:
Sure, I'm super happy and grateful with all the work that @oprypin and others did on Windows, but as we discussed a few months ago, Windows is not a priority for 1.0. We'll focus on filling the missing parts after 1.0. If we introduce alternative constructors or api for Windows, then modifications of that API after 1.0 would be breaking changes. Not big deal anyway, because 1.0 wouldn't be truly available for Windows, so it's not actually a breaking change.

I can add the two initializers, no problem. But I'm sometimes battling with something that is not a priority for 1.0 and making us spend extra valuable time for something that is temporary anyway.

Reply:
Sounds good to me.

Reply:
I mean, it sounds good to drop support for Windows.

Reply:
It's ok, I don't want to drop Windows support 😆 . I just used the approach of two constructors, so now the docs are more accurate and won't change once we fix the issue.